src/pybind/mgr/cephadm/module.py

```diff
@@ -3477,6 +3477,30 @@ def tuned_profile_add_setting(self, profile_name: str, setting: str, value: str)
         self.tuned_profiles.add_setting(profile_name, setting, value)
         self._kick_serve_loop()
         return f'Added setting {setting} with value {value} to tuned profile {profile_name}'
+    
+    @handle_orch_error
+    def tuned_profile_add_settings(self, profile_name: str, settings: dict) -> str:
+        if profile_name not in self.tuned_profiles:
+            raise OrchestratorError(
+                f'Tuned profile {profile_name} does not exist. Cannot add setting.')
+        results = []
+        for key, value in settings.items():
+            self.tuned_profiles.add_setting(profile_name, key, value)
+            results.append(f'Added setting {key} with value {value} to tuned profile {profile_name}')
```

Comment:
I feel like the contents of this function ought to just be a new `add_settings` in the `TunedProfileStore` class. `add_setting` saves the tuned profile after it adds a setting, so repeated calls `add_setting` means a lot of unnecessary save operations.

Reply:
Added a method which takes the dict and processes it in a single run.